### PR TITLE
[TSK-155] setup frontend structure

### PR DESCRIPTION
**Pull request by Hoocki**

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import Providers from "./providers/Providers";
+import {StyledEngineProvider} from "@mui/material";
+
+const App: React.FC = () => {
+
+    return (
+        <StyledEngineProvider injectFirst>
```
Comment (adsukharev) [inline frontend/src/App.tsx]: This provider component should be in providers

File: frontend/src/components/header/mobileLogo/MobileLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import {Typography} from "@mui/material";
+import './styles.css';
+import {Titles} from "../../../common/constansts/titles";
+
+const MobileLogo: React.FC = () => {
+    return (
+        <Typography
+            variant="h5"
+            noWrap
+            component="a"
```
Comment (adsukharev) [inline frontend/src/components/header/mobileLogo/MobileLogo.tsx]: Is "a" really what you want ? I'm just asking, I don't know

File: frontend/src/components/header/mobileMenu/MobileMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'left',
+                }}
+                open={Boolean(anchorEl)}
+                onClose={handleCloseMenu}
+                className="settings-menu"
+            >
+                {HEADER_MAIN_TABS.map((page) => (
+                    <MenuItem key={page} onClick={(event) => handleCloseMenu(event)}>
```
Comment (adsukharev) [inline frontend/src/components/header/mobileMenu/MobileMenu.tsx]:
```suggestion
                    <MenuItem key={page} onClick={handleCloseMenu}>
```
Can you try to do like this?
If it works, this approach is better as you don't need to create additional function, you just pass reference to existing one.
Check other places, where you can a little bit improve performance by using this approach

File: frontend/src/components/header/userMenu/UserMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                    <Avatar/>
+                </IconButton>
+            </Tooltip>
+            <Menu
+                className="user-settings-menu"
+                id="menu-appbar"
+                anchorEl={anchorElUser}
+                anchorOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right',
```
Comment (adsukharev) [inline frontend/src/components/header/userMenu/UserMenu.tsx]: You use the same Params here and below. If you can extract it and reuse,It would be better

**Review by adsukharev — APPROVED** [review]

**Comment by adsukharev** [comment]
Push Kostya to review faster. In the meantime, don't wait and work on the next tickets

File: frontend/src/components/header/mobileLogo/MobileLogo.tsx (hunk 2 above)
Comment (Hoocki) [reply frontend/src/components/header/mobileLogo/MobileLogo.tsx]: no, it isn't necessary, forgot to delete it

**Comment by adsukharev** [comment]
> Push Kostya to review faster. In the meantime, don't wait and work on the next tickets

you can work on the next frontend features. Just create new branch from this one (tsk 155)

**Review by kostya4201337 — APPROVED** [review]

**Pull request merged**